### DISCUSSION TITLE
what is the OpenPose bone color scheme used for the controlnet model?

### DISCUSSION THREAD

**Discussion by if-ai** [lllyasviel/ControlNet#266] [Q&A]
Hi, We have been trying to figure out which is the color scheme that was use to train the openpose model, I tried using the official coco version but somehow seems like the head with the original colors like this yields generations with the subject facing back from the camera most of the time. https://github.com/Mikubill/sd-webui-controlnet/files/10956754/Openpose-18-keypoints_coco_color_codes_v10.pdf It would be very helpful for us to know that information also, thank you so much for making the models. We wish a hand and feet version might one day be available too :D

**Comment by AugmentedRealityCat** [comment]
Here is the most recent version of the Openpose COCO 18 point color reference chart. 
If no errors are found in this version (there were many in v10 linked above that have since been corrected) then I'll publish it more widely and officially. 

[Openpose-18-keypoints_coco_color_codes_v13.pdf](https://github.com/lllyasviel/ControlNet/files/10964930/Openpose-18-keypoints_coco_color_codes_v13.pdf)

![Openpose-18-keypoints_coco_color_codes_v13](https://user-images.githubusercontent.com/114955776/224919175-f9a51fdc-9b77-454c-bdf2-ca68c221abcd.png)

**Comment by Wambosa** [comment]
ID | Joint Name | R | G | B | HEX
-- | -- | -- | -- | -- | --
0 | Nose | 255 | 0 | 0 | #FF0000
1 | Neck | 255 | 85 | 0 | #FF5500
2 | Right Shoulder | 255 | 170 | 0 | #FFAA00
3 | Right Elbow | 255 | 255 | 0 | #FFFF00
4 | Right Wrist | 170 | 255 | 0 | #AAFF00
5 | Left Shoulder | 85 | 255 | 0 | #55FF00
6 | Left Elbow | 0 | 255 | 0 | #00FF00
7 | Left Wrist | 0 | 255 | 85 | #00FF55
8 | Right Hip | 0 | 255 | 170 | #00FFAA
9 | Right Knee | 0 | 255 | 255 | #00FFFF
10 | Right Ankle | 0 | 170 | 255 | #00AAFF
11 | Left Hip | 0 | 85 | 255 | #0055FF
12 | Left Knee | 0 | 0 | 255 | #0000FF
13 | Left Ankle | 85 | 0 | 255 | #5500FF
14 | Right Eye | 170 | 0 | 255 | #AA00FF
15 | Left Eye | 255 | 0 | 255 | #FF00FF
16 | Right Ear | 255 | 0 | 170 | #FF00AA
17 | Left Ear | 255 | 0 | 85 | #FF0055


Joint Pair | Bone Name | R | G | B | HEX
-- | -- | -- | -- | -- | --
1–2 | Right Shoulderblade | 153 | 0 | 0 | #990000
1–5 | Left Shoulderblade | 153 | 51 | 0 | #993300
2–3 | Right Arm | 153 | 102 | 0 | #996600
3–4 | Right Forearm | 153 | 153 | 0 | #999900
5–6 | Left Arm | 102 | 153 | 0 | #669900
6–7 | Left Forearm | 51 | 153 | 0 | #339900
1–8 | Right Torso | 0 | 153 | 0 | #009900
8–9 | Right Upper Leg | 0 | 153 | 51 | #009933
9–10 | Right Lower Leg | 0 | 153 | 102 | #009966
1–11 | Left Torso | 0 | 153 | 153 | #009999
11–12 | Left Upper Leg | 0 | 102 | 153 | #006699
12–13 | Left Lower Leg | 0 | 51 | 153 | #003399
1–0 | Head | 0 | 0 | 153 | 000099
0–14 | Right Eyebrow | 51 | 0 | 153 | #330099
14–16 | Right Ear | 102 | 0 | 153 | #660099
0–15 | Left Eyebrow | 153 | 0 | 153 | #990099
15–17 | Left Ear | 153 | 0 | 102 | #990066
